### PR TITLE
[AFD] adds loading screens

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
adds loading screens

## Why It's Good For The Game
all the real games have them, why not this one?

## Testing
my screen loaded

## Changelog
:cl:
fix: Client optimization by preloading areas. it's society's fault the way this change is
/:cl:

### CODE REVIEW

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -451,6 +453,21 @@
 		if(newarea==oldarea) return
 
 		M.lastarea = src
+		if(istype(M) && M.client)
+			var/loading_times = pick(2 SECONDS, 3 SECONDS, 3.5 SECONDS)
+			if(prob(1))
+				loading_times = 2 MINUTES
```
Comment (AffectedArc07) [inline code/game/area/areas.dm:459]:
```suggestion
				loading_times = 30 SECONDS
```
God please sods law will kick in

Comment (GDNgit) [reply code/game/area/areas.dm:459]: Yeah we should make it take 30 minutes I agree

**Comment by Contrabang** [comment]
I dont understand what this does. Video please?

File: code/game/area/areas.dm [hunk 2]
```diff
@@ -451,6 +453,21 @@
...
+			M.Silence(loading_times)
+			var/atom/movable/screen/loading_screen = new /atom/movable/screen(src)
+			loading_screen.icon = GLOB.title_screen_icon
+			loading_screen.layer = 23
+			loading_screen.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+			loading_screen.screen_loc = "1,1"
+			M.client.screen += loading_screen
+			to_chat(M, "Game is loading!")
+			spawn(loading_times)
+				QDEL_NULL(loading_screen)
```
Comment (Contrabang) [inline code/game/area/areas.dm:470]: needs a span and a timer here

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/area/areas.dm (hunk 2 above)
Comment (GDNgit) [reply code/game/area/areas.dm]: span added, no timer cuz timer spam though

**Comment by Contrabang** [comment]
THE PEOPLE WANT IT GONE
image

**Review by DGamerL — APPROVED** [review]
FIWB

**Pull request merged**